src/components/shared/Button/Button.tsx

```diff
@@ -0,0 +1,58 @@
+import clsx from "clsx";
+import Link from "next/link";
+import ButtonIcon from "./ButtonIcon";
+import { IconName } from "@/types";
+
+export interface ButtonProps {
+  variant: "primary" | "inversion" | "secondary";
+  size: "medium" | "large";
+  icon: boolean;
```

Comment:
Consider defaulting optional props like `icon` and `size` to reduce required boilerplate.